core/src/test/resources/newlines/alwaysBeforeTopLevelStatements.stat

```diff
@@ -144,4 +158,117 @@ object A {
 
   @Singleton
   class B @Inject()(val x: Int)
+
+  @js.native
+  class C() {
+    def c = 5
+  }
+}
+<<< Double annotation
+object a {
+  @annot
+  @bannot
+  class A
+}
+>>>
+object a {
+
+  @annot
+  @bannot
+  class A
+}
+<<< Don't add NL before single-line defs
+object Main {
+  def x = 1
+  def y = 2
+  def z = 3
+
+  def foo = {
+    println("foo")
+  }
+
+  def bar = {
+    println("bar")
+  }
+}
+>>>
+object Main {
+  def x = 1
+  def y = 2
+  def z = 3
+
+  def foo = {
+    println("foo")
+  }
+
+  def bar = {
+    println("bar")
+  }
+}
+
+<<< Add NL before multi-line statements
+object Main {
+  def x = 1
+  def y = 2
+  def z = 3
+  def foo = {
+    println("foo")
+  }
+  def bar = {
+    println("bar")
+  }
+}
+>>>
+object Main {
+  def x = 1
+  def y = 2
+  def z = 3
+
+  def foo = {
+    println("foo")
+  }
+
+  def bar = {
+    println("bar")
+  }
+}
+<<< nested defs (up for debate)
+object a {
+  def foo = {
+    def b = {
+      3
+    }
+    def c = {
+      4
+    }
+  }
+}
+>>>
+object a {
+
+  def foo = {
+
+    def b = {
+      3
+    }
+
+    def c = {
+      4
+    }
+  }
+}
+<<< single line following multiline (up for debate)
+object a {
+  def foo = {
+    2
+  }
+  def bar = 2
+}
+>>>
+object a {
+
+  def foo = {
+    2
+  }
+  def bar = 2
```

Comment:
OK, we can leave that one for a separate PR. I suspect it will not be too hard to accommodate customizations like this one.
 
I'm gonna quickly merge this PR to unblock #898, which moves almost all source files to a new directory layout.